x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -732,6 +734,145 @@ And the Investigation Guide tab should NOT be displayed
 
 TODO: add scenarios https://github.com/elastic/kibana/issues/166215
 
+### Rule upgrade workflow: viewing rule changes before upgrading
+
+#### **Scenario: User can see changes in a side-by-side view**
+
+**Automation**: 1 Cypress e2e test
+
+```Gherkin
+Given X prebuilt rules are installed in Kibana
+And for Y of these rules new versions are available
+When user opens the Rule Updates table and selects a rule
+Then the upgrade preview should open
+And rule changes should be displayed in a two-column view
+And correct rule version numbers should be displayed in their respective columns
+When the user selects another rule without closing the preview
+Then the preview should display the changes for the newly selected rule
+```
+
+#### **Scenario: User can see precisely how property values would change after upgrade**
+
+**Automation**: 1 UI component-level test
+
+```Gherkin
+Given a rule preview with rule changes is open
+Then each line of <column> that was <change_type> should have <bg_color> background
+And marked with <line_badge> badge
+And each changed word in <column> should be highlighted with <accent_color>
```

Comment:
I cannot recall now, but we are only using only WORD diffing for the whole JSON, right? No need to test other uses cases, like character diffing.

Reply:
No, at this stage it's by word only.